install/install_openadapt.ps1

```diff
@@ -31,26 +27,35 @@ $VCRedistRegPath = "HKLM:\SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\X64"
 function RunAndCheck {
     Param
     (
-        [Parameter(Mandatory = $true)] [string] $Command,
-        [Parameter(Mandatory = $true)] [string] $Desc
+        [Parameter(Mandatory = $true)]
+        [string] $Command,
+
+        [Parameter(Mandatory = $true)]
+        [string] $Desc,
+
+        [Parameter(Mandatory = $false)]
+        [switch] $SkipCleanup = $false
     )
 
     Invoke-Expression $Command
     if ($LastExitCode) {
-        Write-Host "Failed: $Desc : $LastExitCode"
-        Cleanup
-        exit
+        Write-Host "Failed: $Desc - Exit code: $LastExitCode" -ForegroundColor Red
+        if (!$SkipCleanup) {
+            Cleanup
+            exit
+        }
+    }
+    else {
+        Write-Host "Success: $Desc" -ForegroundColor Green
     }
-
-    Write-Host "Success: $Desc"
 }
 
 
 function Cleanup {
-    $exists = Test-Path -Path "..\OpenAdapt"
+    $exists = Test-Path -Path $setupdir
```

Comment:
Where does C:/OpenAdaptSetup come from? What if they download OpenAdapt in a different folder? When git clone is run on this repository, doesn't it create a folder called OpenAdapt?

Reply:
The reason I have OpenAdaptSetup folder is that in Windows when we run the script with Administrative rights it runs in system32 folder by default. And we can not unpack an installer using commandline in that folder as it is protected. Hence, I `cd` to a new directory that is out of system32 and then I run all of the commands there. It was a fix for NSIS error for windows.

Reply:
I hope this clarified! It is upto you to have a similar setup. :)